### DISCUSSION TITLE
FolderBrowserDialog not opening in fsharp script

### DISCUSSION THREAD

**Discussion by kevinhayesanderson** [dotnet/fsharp#14110] [General]
I am having a simple script to select a folder, 
the process is not responding on call to ShowDialog(), 
below are script file, runtime config file and cli command to run,
I think i'm missing something or my understanding is wrong, 

test.fsx
`
#I @"C:\Program Files\dotnet\packs\Microsoft.WindowsDesktop.App.Ref\6.0.10\ref\net6.0" 
#r "System.Windows.Forms"

open System
open System.Windows.Forms

let folderPath = new Label()
let button = new Button(Text = "Select Folder")
let ini_dir = @"C:\Users\hayeskev\source\Workspaces"

let openFolderHandler = 
    EventHandler(fun _ _ ->
        let folderDlg = new FolderBrowserDialog()
        folderDlg.InitialDirectory <- ini_dir
        folderDlg.ShowNewFolderButton <- true
        let dlgResult = folderDlg.ShowDialog()
        if dlgResult.Equals(DialogResult.OK) then 
            folderPath.Text <- folderDlg.SelectedPath
        else
            folderPath.Text <- "Error")

button.Click.AddHandler(openFolderHandler)

let layout = new FlowLayoutPanel(Dock=DockStyle.Fill)
layout.Controls.Add(button)
layout.Controls.Add(folderPath)
let form = new Form()
form.Controls.Add(layout)
form.ShowDialog()
`

C:\Program Files\dotnet\sdk\6.0.402\FSharp\fsi.runtimeconfig.json
`
{
  "runtimeOptions": {
    "tfm": "net6.0",
    "frameworks": [
      {
        "name": "Microsoft.NETCore.App",
        "version": "6.0.10"
      },
      {
        "name": "Microsoft.WindowsDesktop.App",
        "version": "6.0.10"
      }
    ],
    "configProperties": {
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": false
    }
  }
}
`
running using 
`
dotnet fsi test.fsx
`
![image](https://user-images.githubusercontent.com/22426020/195678761-dc1da2b6-f4f7-4d0a-80ff-10053f2a0956.png)
![image](https://user-images.githubusercontent.com/22426020/195679017-e4592630-22ac-43c3-9309-ac2129b3d677.png)

**Comment by vzarytovskii** [comment]
I think it's the same issue as mentioned in 

https://github.com/dotnet/fsharp/issues/12023

Here's the comment describing what's happening in details: https://github.com/dotnet/fsharp/issues/12023#issuecomment-1224947800
